src/rm_tdigest.c

```diff
@@ -389,6 +389,73 @@ static int double_cmpfunc(const void *a, const void *b) {
 }
 #endif
 
+/**
+ * Command: TDIGEST.RANK {key} {value} [{value}...]
+ *
+ * Retrieve the estimated rank of value
+ * (the number of observations in the sketch that are smaller than value +
+ * half the number of observations that are equal to value)
+ *
+ * @param ctx Context in which Redis modules operate
+ * @param argv Redis command arguments, as an array of strings
+ * @param argc Redis command number of arguments
+ * @return REDISMODULE_OK on success, or REDISMODULE_ERR  if the command failed
+ */
+int TDigestSketch_Rank(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc < 3) {
+        return RedisModule_WrongArity(ctx);
+    }
+
+    RedisModuleString *keyName = argv[1];
+    RedisModuleKey *key = RedisModule_OpenKey(ctx, keyName, REDISMODULE_READ);
+
+    if (_TDigest_KeyCheck(ctx, key) != REDISMODULE_OK)
+        return REDISMODULE_ERR;
+
+    td_histogram_t *tdigest = RedisModule_ModuleTypeGetValue(key);
+    if(td_size(tdigest) == 0) {
+        RedisModule_ReplyWithArray(ctx, 1);
+        RedisModule_ReplyWithDouble(ctx, NAN);
+        RedisModule_CloseKey(key);
+        return;
+    }
+
+    const size_t n_values = argc - 2;
+    double *vals = (double *)__td_calloc(n_values, sizeof(double));
+
+    for (int i = 0; i < n_values; ++i) {
+        if ((RedisModule_StringToLongLong(argv[2 + i], &vals[i]) != REDISMODULE_OK) || isnan(vals[i])) {
+            RedisModule_CloseKey(key);
+            __td_free(vals);
+            return RedisModule_ReplyWithError(ctx, "ERR T-Digest: error parsing value");
+        }
+    }
+
+    int *ranks = (int *)__td_calloc(n_values, sizeof(int));
+
+    double size = td_size(tdigest);
+    double min = td_min(tdigest);
+    double max = td_min(tdigest);
+    for (int i = 0; i < n_values; ++i) {
+        if(vals[i] < min) {
+            ranks[i] = -1;
+        } else if (vals[i] > max) {
+            ranks[i] = td_size(tdigest);
```

Comment:
What does `-1` or `td_size` symbol?

Reply:
I've updated the PR description to include the command behavior:
 (-1 when value < value of the smallest observation, n when value > value of the largest observation).